lib/ansible/modules/cloud/amazon/ec2_elb_lb.py

```diff
@@ -738,10 +738,24 @@ def _create_elb(self):
             self.status = 'created'
 
     def _create_elb_listeners(self, listeners):
-        """Takes a list of listener tuples and creates them"""
-        # True if succeeds, exception raised if not
-        self.changed = self.elb_conn.create_load_balancer_listeners(self.name,
-                                                                    complex_listeners=listeners)
+        """Takes a list of listener tuples and creates them
+
+        Arguments:
+            listeners (list): list of tuples
+
+        Returns:
+            True if succeeds
+
+        Raises:
+            IndexError: If HTTPS is specified as a listener but SSL cert is not specified.
+
+        """
+        try:
+            self.changed = self.elb_conn.create_load_balancer_listeners(self.name,
+                                                                        complex_listeners=listeners)
+        except IndexError as e:
+            self.module.fail_json(msg='unable to create listener for port 443 HTTPS without specifying an SSL cert: %s'
```

Comment:
You don't need to hardcode the port.

Reply:
msg='unable to create listener for HTTPS without specifying an SSL cert: %s'

Is that better?